### ISSUE TITLE
de-globalization?

### ISSUE THREAD

**Issue by sgtlaggy** [dis446/Fallout-Shelter-Text-Based#16]
Should this be worked on or no?

**Comment by dis446** [comment]
Let me handle that aspect.

**Comment by dis446** [comment]
But yeah de-globalisation would be good.

**Comment by kungtotte** [comment]
I wrote up an example on how OOP and de-globalization could be done. It's not complete in any way and really only focuses on the main stuff of the game, but the same principle could be applied everywhere in the code.

In general the main problems I find in this game are the over-reliance on global variables and global state, as well as the penchant for using strings and indexes to store things instead of using the many proven data structures that Python provides like lists, dictionaries, sets, namedtuples and so on. Right now it's pretty hard to follow and it's all very interconnected so that changing one little piece affects a lot of other things.

I think it would be worthwhile to sketch out a general design of the code first before trying to refactor it, and here's one suggestion for how to do it.

[Link to code here](https://gist.github.com/kungtotte/0e685972674f0ef21795)

**Comment by sgtlaggy** [comment]
That's really good, I think that would be the best way to go about this.

**Comment by dis446** [comment]
This is really awesome. I would like us all to work on this actually. I'll make a new branch and let's focus on reforming the code to make this happen!

**Comment by dis446** [comment]
Looking over the code, the global issue seems to have been solved at this point.

**Comment by sgtlaggy** [comment]
I wouldn't say it's quite solved yet. Every time I load up `Human.py` and `Room.py`, I get undefined warnings for variables `people` and `person`, but I think that might be all that's left.
